packages/@o3r/design/src/core/design-token/renderers/sass/design-token-definition.renderers.ts

```diff
@@ -0,0 +1,34 @@
+import type { DesignTokenVariableStructure, TokenKeyRenderer, TokenValueRenderer } from '../../parsers/design-token-parser.interface';
+import type { TokenDefinitionRenderer } from '../design-token.renderer.interface';
+import { getCssTokenValueRenderer } from '../css/design-token-value.renderers';
+
+interface SassTokenDefinitionRendererOptions {
+  /** Custom Design Token value renderer */
+  tokenValueRenderer?: TokenValueRenderer;
+
+  /**
+   * Renderer the name of the Sass Variable (without initial $)
+   * @default {@see fromCssVarNameToSassVar}
+   **/
+  tokenVariableNameRenderer?: TokenKeyRenderer;
+}
+
+const fromCssVarNameToSassVar: TokenKeyRenderer = (variable) => {
+  const tokens = variable.getKey().split('-');
+  return tokens[0] + tokens.slice(1).map((token) => token.charAt(0).toUpperCase() + token.slice(1)).join('');
+};
+
+/**
+ * Retrieve the Design Token Variable renderer for Sass
+ * @param options
+ * @returns
+ */
+export const getSassTokenDefinitionRenderer = (options?: SassTokenDefinitionRendererOptions): TokenDefinitionRenderer => {
```

Comment:
Just to make sure I did not miss something: this is only to support a way to hide private tokens from the bundle css?

Reply:
That could be nice to document it

Reply:
This is actually the opposite.
It's a render for Design Token that will render it as Sass variable.
It can be provided to the CSS render to provide a way to render private variable but it is not mandatory at all and not the only way to render private variable.

Per default the Private Variable are not rendered by the CSS Render as specified in the interface:
```typescript
  /**
   * Private Design Token definition renderer
   * The private variable will not be rendered if not provided
   */
  privateDefinitionRenderer?: TokenDefinitionRenderer;
```

Do you have an idea of addition documentation that can be provided?

Reply:
As discussed, it can be handled in https://github.com/AmadeusITGroup/otter/issues/1126